recipes/game-networking-sockets/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class GameNetworkingSocketsConan(ConanFile):
+    name = "game-networking-sockets"
+    description = "GameNetworkingSockets is a basic transport layer for games."
+    topics = ("networking", "game-development")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ValveSoftware/GameNetworkingSockets"
+    license = "BSD 3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "encryption": ["openssl", "libsodium", "bcrypt"]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "encryption": "openssl"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.encryption == "bcrypt" and self.settings.os != "Windows":
+            raise Invalid()
```

Comment:
For unsupported configurations, raise a `ConanInvalidConfiguration` in `validate()` like that https://github.com/conan-io/conan-center-index/blob/3d96da9fe3db0bf8a24325e7cad1b4f1b81d4f23/recipes/sdl_ttf/all/conanfile.py#L52